File: csharp/1227.airplane-seat-assignment-probability.cs
/*
 * @lc app=leetcode id=1227 lang=csharp
 *
 * [1227] Airplane Seat Assignment Probability
 */

// @lc code=start
public class Solution {
    public double NthPersonGetsNthSeat(int n) {
        return n == 1 ? 1 : 0.5;
    }
}
// @lc code=end

File: java/1227.airplane-seat-assignment-probability.java
/*
 * @lc app=leetcode id=1227 lang=java
 *
 * [1227] Airplane Seat Assignment Probability
 */

// @lc code=start
class Solution {
    public double nthPersonGetsNthSeat(int n) {
        if (n == 1) {
            return 1.0;
        }
        return 1 / (double)n + (double)(n - 2) / (double)n * nthPersonGetsNthSeat(n - 1);
    }
}
// @lc code=end

